zcash_client_sqlite/src/wallet.rs

```diff
@@ -2120,40 +2114,34 @@ pub(crate) fn get_received_memo(
     conn: &rusqlite::Connection,
     note_id: NoteId,
 ) -> Result<Option<Memo>, SqliteClientError> {
-    let fetch_memo = |table_prefix: &'static str, output_col: &'static str| {
-        conn.query_row(
+    let TableConstants {
+        table_prefix,
+        output_index_col,
+        ..
+    } = table_constants::<SqliteClientError>(note_id.protocol())?;
+
+    let memo_bytes = conn
+        .query_row(
             &format!(
                 "SELECT memo FROM {table_prefix}_received_notes
                 JOIN transactions ON {table_prefix}_received_notes.tx = transactions.id_tx
                 WHERE transactions.txid = :txid
-                AND {table_prefix}_received_notes.{output_col} = :output_index"
+                AND {table_prefix}_received_notes.{output_index_col} = :output_index"
             ),
             named_params![
                 ":txid": note_id.txid().as_ref(),
                 ":output_index": note_id.output_index()
             ],
-            |row| row.get(0),
+            |row| row.get::<_, Option<Vec<u8>>>(0),
         )
-        .optional()
-    };
+        .optional()?
+        .flatten();
```

Comment:
This looks like an undocumented bugfix: it converts an error if the received note's tx is unenhanced (because `null` can't be parsed into `Vec<u8>`) into `Ok(None)`. Update the method documentation to clarify this, and add a "Fixed" entry to the `zcash_client_sqlite` changelog (because this alters the implementation to match the documented behaviour of `WalletRead::get_memo`).

Reply:
Ah okay, so previously the `memo_bytes: Option<Vec<_>>` combined with the `.flatten()` was enough for Rust to figure out that the return type of `fetch_memo` was `Result<Option<Option<Vec<_>>>, _>`. I've confirmed this in current `main`.